tpm2/tpm2.go

```diff
@@ -1546,6 +1546,26 @@ func encodeCertify(parentAuth, ownerAuth string, object, signer tpmutil.Handle,
 	return concat(ha, auth, params)
 }
 
+// This function differs from encodeCertify in that it takes the scheme
+// to be used as an additional argument.
+func encodeCertifyEx(parentAuth, ownerAuth string, object, signer tpmutil.Handle, qualifyingData tpmutil.U16Bytes, scheme tpmtSigScheme) ([]byte, error) {
+	ha, err := tpmutil.Pack(object, signer)
+	if err != nil {
+		return nil, err
+	}
+
+	auth, err := encodeAuthArea(AuthCommand{Session: HandlePasswordSession, Attributes: AttrContinueSession, Auth: []byte(parentAuth)}, AuthCommand{Session: HandlePasswordSession, Attributes: AttrContinueSession, Auth: []byte(ownerAuth)})
+	if err != nil {
+		return nil, err
+	}
+
+	params, err := tpmutil.Pack(qualifyingData, scheme)
```

Comment:
Can you update this to correctly pack TPM_ALG_NULL (with no following hash alg) for the NULL case of TPMT_SCHEME? Feel free to file a GitHub issue for "tpmtSigScheme does not match TPMT_SIG_SCHEME" and reference it with a TODO comment explaining that this special-case command packing would be a workaround and not needed after the bug is fixed.

Reply:
Do you mean when the passed `scheme` is `nil` ? or when the `Scheme` of `tpmtSigScheme` is `AlgNull` ? or both ?

Reply:
I think either one would be workable.

My 2 cents: I prefer continuing to take `scheme` by value, and special-casing `if scheme.Scheme == tpmAlgNull` as a workaround (making sure to reference the GitHub issue in a comment, so we can clean it up later when fixed), as personally I might wonder as a user if the API will substitute some default value if I provide `nil` for `scheme`. AlgNull is very clear and helps guide the user to expectations based in the TPM spec.

Reply:
Does this look good now ?